api/src/main/java/io/opencensus/metrics/export/MetricProducerManager.java

```diff
@@ -79,11 +56,23 @@ public synchronized void remove(MetricProducer metricProducer) {
    *
    * @return all registered {@code MetricProducer}s that should be exported.
    */
-  public Set<MetricProducer> getAllMetricProducer() {
-    return metricProducers;
-  }
+  public abstract Set<MetricProducer> getAllMetricProducer();
+
+  static final class NoopMetricProducerManager extends MetricProducerManager {
```

Comment:
this should be private. You should add a "public static MetricProducerManager newNoopMetricProducerManager() {}" to be consistent with the other APIs (like stats/trace).

Reply:
Got it, updated.